templates/gitlab/mender-deployment-create.gitlab-ci.yml

```diff
@@ -0,0 +1,62 @@
+# Use this template to create deployments for Mender artifacts from MENDER_ARTIFACTS_FOLDER folder.
+#
+# The template is designed to be used with the includes keyword or copied into an existing
+# configuration. It requires `MENDER_SERVER_URL` and `MENDER_SERVER_ACCESS_TOKEN` variables
+# to work and can be used to create a deployment on a Mender server.
+#
+# It's possible to set custom values by setting the following environment variables:
+# - `MENDER_DEPLOYMENT_GROUP`: The name of Mender devices group a deployment will be triggered to.
+#                              One of `MENDER_DEPLOYMENT_GROUP` or `MENDER_DEVICES_LIST` is required.
+# - `MENDER_DEVICES_LIST`: The list of Mender devices a deployment will be triggered to.
+# - `MENDER_DEPLOYMENT_NAME` (optional): The name of Mender deployment.
+#                                        (default: `${RELEASE_NAME}_${CI_COMMIT_SHORT_SHA}`)
+#
+# Deployment group/name and devices list can be set by placing its values into GitLab CI artifacts
+# folder into files with related environment variables as suffix.
+# (ex. `MENDER_ARTIFACTS_FOLDER/${ARTIFACT_NAME}MENDER_DEPLOYMENT_GROUP`)
+
+include:
+  - remote: "https://raw.githubusercontent.com/mendersoftware/mender-ci-workflows/master/templates/gitlab/common.gitlab-ci.yml"
+
+.create:mender:deployment:
+  image: mendersoftware/mender-ci-tools:master
+  variables:
+    MENDER_DEPLOYMENT_GROUP: "" # The name of Mender devices group a deployment will be triggered to. One of `MENDER_DEPLOYMENT_GROUP` or `MENDER_DEVICES_LIST` is required
+    MENDER_DEVICES_LIST: ""     # The list of Mender devices a deployment will be triggered to
+    MENDER_DEPLOYMENT_NAME: ""  # (optional): The name of Mender deployment (default: `${RELEASE_NAME}_${CI_COMMIT_SHORT_SHA}`)
+  script:
+    - for ARTIFACT in $(ls ${MENDER_ARTIFACTS_FOLDER}/*.mender); do
+    -   'echo "Processing artifact ${ARTIFACT}"'
+    -   RELEASE_NAME=$(mender-artifact read --no-progress ${ARTIFACT} | grep "Name:" | awk {'print $2'})
```

Comment:
I understand why we do this, but in theory this job should not need to do anything with the Mender Artifact. We should pass the release name through a env variable. See [here](https://docs.gitlab.com/ee/ci/variables/#pass-an-environment-variable-to-another-job).

Reply:
Then we need to review the concept of processing more than one artifact in a job in general.

If it's going to process more than one artifact then it needs to have a way of transferring more than one `RELEASE_NAME` and `MENDER_DEVICES_GROUP_NAME`. Receiving an array with those values via environment variables doesn't look attractive to me (because of a need to synchronise IDs in the arrays with the list of files).